cl/api/utils.py

```diff
@@ -450,6 +454,115 @@ def list(self, *args, **kwargs):
+    """
+    return (
+        f"{prefix}_count" if is_count_request else f"{prefix}_{ordering_key}"
+    )
+
+
+class NoFilterCacheListMixin:
+    """
+    A mixin that caches list of results when there's no pagination and either a
+    count is requested or no filters are applied to the queryset.
+
+    It leverages Django's caching mechanism to store responses when no filters
+    or pagination are applied to the queryset, improving performance by
+    avoiding repeated database queries for frequently accessed data.
+
+    Attributes:
+        no_filters_cache_key (str, optional): A custom prefix for the cache key.
+            If not provided, the class name will be used as the prefix.
+    """
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+        has_filters = queryset.query.has_filters()
+        is_count_request = (
+            request.query_params.get("count") == "on"
+            and request.version == "v4"
+        )
+        has_pagination = request.query_params.get(
+            "cursor"
+        ) or request.query_params.get("page")
+        is_v3_request = request.version == "v3"
+
+        # Determine the cache key prefix. Uses a custom key if provided,
+        # otherwise default to the class name.
+        prefix = getattr(self, "no_filters_cache_key", self.__class__.__name__)
+        # Get the ordering key from the request, defaulting to the view's
+        # ordering
+        ordering_key = request.query_params.get("order_by", self.ordering)
+
+        cache = caches["db_cache"]
+        cache_key = get_cache_key_for_no_filter_mixin(
+            prefix, ordering_key, is_count_request
+        )
+
+        # Attempt to retrieve the response from cache only if eligible.
+        # Caching is applied when there's no pagination and either a count is
+        # requested or no filters are active, ensuring we cache full,
+        # unfiltered/unpaginated datasets or counts, which are likely to be
+        # reused frequently.
+        should_cache_response = (
+            not is_v3_request
+            and not has_pagination
+            and (is_count_request or not has_filters)
+        )
+        if should_cache_response:
+            response = cache.get(cache_key) or None
+            if response:
+                return response
+
+        def _save_page_in_cache(
+            cache_connection: BaseCache,
+            key: str,
+            response: SimpleTemplateResponse,
+        ):
+            """
+            Helper function to save the response in the cache.
+
+            Args:
+                cache_connection: The cache instance (e.g., Django's `caches["db_cache"]`).
+                key (str): The cache key under which to store the response.
+                response_obj (Response): The DRF Response object to be cached.
+            """
+            # Cache the response for 1 hour
+            cache_connection.set(key, response, 60 * 60)
```

Comment:
I think we can do ten minutes here:

```suggestion
            cache_connection.set(key, response, 10 * 60)
```

Reply:
Sure! I'll update the timeout.